tests/repos_events_test.rs

```diff
@@ -0,0 +1,143 @@
+// Tests for calls to the /repos/{owner}/{repo}/events API.
+use octocrab::{
+    etag::{Etag, Etagged},
+    models::events,
+    Octocrab, OctocrabBuilder,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::convert::TryFrom;
+use wiremock::{
+    matchers::{method, path, path_regex},
+    Mock, MockServer, ResponseTemplate,
+};
+
+#[derive(Serialize, Deserialize)]
+struct FakePage<T> {
+    items: Vec<T>,
+}
+
+async fn setup_api(template: ResponseTemplate) -> MockServer {
+    let owner = "owner";
+    let repo = "repo";
+    let mock_server = MockServer::start().await;
+    Mock::given(method("GET"))
+        .and(path(format!("/repos/{}/{}/events", owner, repo)))
+        .respond_with(template)
+        .mount(&mock_server)
+        .await;
+    setup_error_handler(
+        &mock_server,
+        &format!("GET on /repo/{}/{}/events was not received", owner, repo),
+    )
+    .await;
+    mock_server
+}
+
+// Sets up a handler on the mock server which will return a 500 with the given message. This
+// will be mapped internally into a GitHub json error, making it much easier to identify the cause
+// of these test failures.
+//
+// This handler should always come after your real expectations as it will match any GET request.
+async fn setup_error_handler(mock_server: &MockServer, message: &str) {
+    Mock::given(method("GET"))
+        .and(path_regex(".*"))
+        .respond_with(ResponseTemplate::new(500).set_body_json(json!( {
+            "documentation_url": "",
+            "errors": None::<Vec<serde_json::Value>>,
+            "message": message,
+        })))
+        .mount(&mock_server)
+        .await;
+}
+
+fn setup_octocrab(uri: &str) -> Octocrab {
+    OctocrabBuilder::new()
```

Comment:
Minor nit: You can just call `Octocrab::builder` which would remove the need to import `OctocrabBuilder`, but it's not super important.
```suggestion
    Octocrab::builder()
```